msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java

```diff
@@ -292,4 +118,11 @@ void acquireTokenSilentAsync(@NonNull final String[] scopes,
      * @return
      */
     PublicClientApplicationConfiguration getConfiguration();
+
+    /**
+     * Returns whether the application is being run on a device that is marked as a shared.
+     * Only SingleAccountPublicClientApplications may be used on shared devices
+     * @return
+     */
+    Boolean getIsSharedDevice();
```

Comment:
Regarding naming, I'd be fine with this being `isSharedDevice()` rather than an explicit getter